app/api/schema/discount_codes.py

```diff
@@ -90,6 +92,36 @@ def validate_quantity(self, data, original_data):
                 raise UnprocessableEntity({'pointer': '/data/attributes/tickets-number'},
                                           "tickets-number should be greater than max-quantity")
 
+    @validates_schema(pass_original=True)
+    def validate_value(self, data, original_data):
+        if 'id' in original_data['data']:
+            try:
+                discount_code = DiscountCode.query.filter_by(id=original_data['data']['id']).one()
+            except NoResultFound:
+                raise ObjectNotFound({'parameter': '{code}'}, "DiscountCode: not found")
+
+            if 'type' not in data:
+                data['type'] = discount_code.type
+
+            if 'value' not in data:
+                data['value'] = discount_code.value
+
+        if data['type'] == "percent":
+            if data['value'] > 100:
```

Comment:
Why isn't this handled in Marhsmallow schema?

Reply:
@iamareebjamal actually the `value` field holds both the `percentage` and `amount` of discount depending on user selection. So I don't think we can restrict this to schema itself bcoz `value` can have more than 100 when it depicts the `amount`.

Reply:
OK, handle negative values as well